### DISCUSSION TITLE
Svelte (v2) queryStore not updating after mutation

### DISCUSSION THREAD

**Discussion by freb** [urql-graphql/urql#2503] [Q&A] [answered]
As far as I understand it, when you create a `queryStore`, its `data` attribute should automatically update whenever one of the nodes in its data was updated. So, for example, If I create a `queryStore` to fetch a list of users to display on the page, if I run a mutation to update the `firstName` field on one of the users in that queryStore result, returning the id and firstName field of that user, the `queryStore` should update to reflect the updated `firstName` field because the cache was updated.

In my testing, I have not been able to get a `queryStore` result to update based on a mutation. Am I understanding this wrong, or is something just going wrong? If the latter, what is the best way to create a reproduction to show the issue?

**Comment by JoviDeCroock** [comment]
Hard to help without info on what you are doing.

Doc-cache should refetch the query (network tab)
Norm-cache should synchronously update

The essence here being typename matches, a stackblitz or codesandbox should be sufficient to reproduce or just additional info. I have less experience with our current bindings so I have not tested it

**Reply by freb** [reply]
@JoviDeCroock, thanks for the reply.

I'm using normalized cache and have created a reproduction here:

https://stackblitz.com/edit/vitejs-vite-2ssvxj?file=src/routes/index.svelte

As you'll see, I have one queryStore and one mutationStore. When the mutation runs (upon clicking SAVE), the queryStore doesn't update even though they are both dealing with the same typename (User) with the same id. I tried adding a resolver to help with this, but it did not work and I didn't think it would be required in this case anyway.

Any light you can shed would be much appreciated.

**Reply by JoviDeCroock** [reply]
Seems like a bug in the svelte bindings

**Reply by freb** [reply]
Being my first use of urql, I wasn't 100% sure on the expected behaviour. So, from what you know, the queryStore should be updated as a result of the mutation? Should I open an issue?

**Reply by mpiorowski** [reply]
@JoviDeCroock I have the same problem as freb, mutationStore not updating queryStore, same typename and id

**Comment by mpiorowski** [comment]
@freb Have You manage to make it work somehow?

**Reply by freb** [reply]
@mpiorowski No, there appears to be a bug in the new Svelte v2 bindings. I'm planning on opening an issue, but was waiting for @JoviDeCroock to reply. I'll open it today.

Based on @jonathanstanley 's reply [here](https://github.com/FormidableLabs/urql/discussions/2460#discussioncomment-2960772), it looks like the v2 bindings were merged prematurely. I actually really like the v2 API now that I know how to use it. So I'm hoping it will get fixed, but I guess the maintainers will have to figure out if they want to fix it or revert back to v1.

Unfortunately, I'm not great with JavaScript or I would try to offer some help.

**Reply by JoviDeCroock** [reply]
We want to fix it, just finding time 😅 sorry.... Will try later

**Reply by freb** [reply]
Hey, @JoviDeCroock, thanks for the reply!. No rush from me. I did go ahead and open the issue to track progress, but confirmation that you plan to move forward with v2 is good enough for me at the moment. Again, thanks for the wonderful project.

**Reply by mpiorowski** [reply]
same here, i rly like the new api :) amazing job guys

**Reply by JoviDeCroock** [reply]
Fixed in #2514

**Comment by freb** [comment, marked as answer]
This was answered [here](https://github.com/FormidableLabs/urql/discussions/2503#discussioncomment-2989151). It was fixed in `@urql/svelte` release `2.0.2`.
